core-modules/000QubesVm.py

```diff
@@ -345,7 +346,14 @@ def __init__(self, **kwargs):
                 self.services['qubes-update-check'] = False
 
         # Initialize VM image storage class
-        self.storage = defaults["storage_class"](self)
+        self.storage = get_pool(self.pool_name, self).getStorage()
+        self.dir_path = self.storage.vmdir
+        self.icon_path = os.path.join(self.storage.vmdir, 'icon.png')
+        self.conf_file = os.path.join(self.storage.vmdir, self.name + '.conf')
+        self.appmenus_templates_dir = os.path.join(self.storage.vmdir, 'apps.templates')
+        self.appmenus_icons_dir = os.path.join(self.storage.vmdir, 'apps.icons')
```

Comment:
I think those two lines should be moved to the appmenus plugin. Maybe change that to some properties with getter calculating the path at access time (to not worry about self.dir_path not set yet).